nix_update/eval.py

```diff
@@ -142,13 +142,22 @@ def eval_expression(
     return f"""
 let
   {indent(dedent(let_bindings), "  ")}
+  positionFromMeta = pkg: let
+    parts = builtins.match "(.*):([0-9]+)" pkg.meta.position;
+  in {{
+    file = builtins.elemAt parts 0;
+    line = builtins.fromJSON (builtins.elemAt parts 1);
+  }};
+
   raw_version_position = sanitizePosition (builtins.unsafeGetAttrPos "version" pkg);
 
-  position = if pkg ? isRubyGem then
+  position = if pkg ? meta.position then
+    sanitizePosition (positionFromMeta pkg)
+  else if pkg ? isRubyGem then
     raw_version_position
   else if pkg ? isPhpExtension then
     raw_version_position
-   else
+  else
     sanitizePosition (builtins.unsafeGetAttrPos "src" pkg);
```

Comment:
Indeed if it is possible to set `meta.position`, this `nix-update` PR here makes a bit more sense. However, since `mkDerivation` always sets `meta.position`, the `pkg.src` attribute will never be retrieved here. Or perhaps it will be retrieved for other derivation builders? I doubt if there are builders that are not based on `mkDerivation` for which it will happen.